### PR TITLE
Cleaning and re-writing of selection tool

### LINKED ISSUE #52: Re-write select restriction tool
This tool is causing random map freeze, and sometime only works in small spots over the restrictions (see #42)

### PR DESCRIPTION
**Pull request by Djedouas**
## Fixes #52 

The selection tool image has been completely re-written using QGIS native functionalities.

Its code is stabilized, more compact, and easier to maintain.

## Cleaning

Mixin classes which were not used correctly have been deleted, and the code has been migrated to standard functions in a `utils.py` file.

Functions related to camera/photo have been removed.

Dead code has been found and removed.

### CODE REVIEW

File: TOMsPlugin/core/tomsProposal.py [hunk 1]
```diff
@@ -20,13 +20,13 @@
...
 from ..constants import ProposalStatus, RestrictionAction
-from ..proposalTypeUtilsClass import ProposalTypeUtilsMixin
+from ..utils import getRestrictionLayersList
 from .tomsMessageLog import TOMsMessageLog
 from .tomsProposalElement import ProposalElementFactory, TOMsProposalElement
 from .tomsTile import TOMsTile
 
 
-class TOMsProposal(ProposalTypeUtilsMixin, QObject):
+class TOMsProposal(QObject):
```
Comment (benoitdm-oslandia) [inline TOMsPlugin/core/tomsProposal.py:29]: add docstring for new or modified classes

File: TOMsPlugin/core/proposalsManager.py [hunk 2]
```diff
@@ -18,13 +18,13 @@
 from qgis.PyQt.QtCore import QDate, QObject, pyqtSignal
```
Comment (benoitdm-oslandia) [inline TOMsPlugin/core/proposalsManager.py:18]: For each modified file we should check a header exists if not add a new one.

File: TOMsPlugin/generateGeometryUtils.py [hunk 3]
```diff
@@ -29,6 +29,7 @@
 from qgis.utils import iface
 
 from .core.tomsMessageLog import TOMsMessageLog
+from .utils import getLookupDescription
 
 
 class GenerateGeometryUtils(QObject):
```
Comment (benoitdm-oslandia) [inline TOMsPlugin/generateGeometryUtils.py:35]: docstring

File: TOMsPlugin/utils.py [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+    QgsProject,
+    QgsSettings,
+)
+from qgis.PyQt.QtWidgets import QDockWidget, QMessageBox
+from qgis.utils import iface
+
+from .core.tomsMessageLog import TOMsMessageLog
+
+
+def restrictionInProposal(currRestrictionID, currRestrictionLayerID, proposalID):
```
Comment (benoitdm-oslandia) [inline TOMsPlugin/utils.py:14]: header and doctring for each function please

**Review by benoitdm-oslandia — CHANGES_REQUESTED** [review]
For each modified file we should check a header exists if not add a new one.

The same for python classes without docstring we need to modify, we should create a new docstring.

What do you think?

**Comment by Djedouas** [comment]
I added some comments, and where there is no comment, it's because I don't really know what or why it's doing.

**Review by benoitdm-oslandia — APPROVED** [review]

**Pull request merged**